retrieval/raw/raw_endpoint.go

```diff
@@ -114,13 +124,145 @@ func downloadToFile(c *Client, method, sourceName string, workDir *os.File, URL
 		return filename, errors.New("Unable to create raw metric file")
 	}
 	defer util.SafeClose(rawRespFile.Close, &rerr)
-
 	filename = rawRespFile.Name()
 
+	if _, ok := ParsableFileSet[sourceName]; c.parseMetricData && ok {
+		err = parseAndWriteData(sourceName, resp.Body, rawRespFile)
+		return filename, err
+	}
+
 	_, err = io.Copy(rawRespFile, resp.Body)
 	if err != nil {
 		return filename, errors.New("Error writing file: " + rawRespFile.Name())
 	}
 
 	return filename, rerr
 }
+
+// TODO: investigate streamed json reading / writing
+func parseAndWriteData(filename string, reader io.Reader, writer io.Writer) error {
+	var to = getType(filename)
+	out := reflect.New(reflect.TypeOf(to))
+	err := json.NewDecoder(reader).Decode(out.Interface())
+
+	if err != nil {
+		return err
+	}
+	to = sanitizeData(out.Elem().Interface())
+
+	data, err := json.Marshal(to)
+	if err != nil {
+		return err
+	}
+	_, err = io.Copy(writer, bytes.NewReader(data))
+	return err
```

Comment:
Do you think we need add some error information to help if err happens? Like unable to parse which file, not supported type and whatever.